dask/array/fft.py

```diff
@@ -116,18 +143,56 @@ def fft_wrap(fft_func, kind=None, dtype=None):
 
     _dtype = dtype
 
-    def func(a, n=None, axis=-1):
-        dtype = _dtype
-        if dtype is None:
-            dtype = fft_func(np.ones(8, dtype=a.dtype)).dtype
+    if kind.endswith("2") or kind.endswith("n"):
+        _kind = kind
+
+        def func(a, s=None, axes=None):
+            kind = _kind
+            dtype = _dtype
+
+            if axes is None:
+                if kind.endswith("2"):
+                    axes = (-2, -1)
+                elif kind.endswith("n"):
+                    if s is None:
+                        axes = tuple(range(a.ndim))
+                    else:
+                        axes = tuple(range(len(s)))
+
+            if dtype is None:
+                dtype = fft_func(np.ones(len(axes) * (8,),
+                                         dtype=a.dtype)).dtype
+
+            for each_axis in axes:
+                if len(a.chunks[each_axis]) != 1:
+                    raise ValueError(
+                        chunk_error % (each_axis, a.chunks[each_axis]))
+
+            chunks = out_chunk_fn(a, s, axes)
+
+            return a.map_blocks(fft_func, s, axes, dtype=dtype,
+                                chunks=chunks)
+    else:
+
+        def func(a, n=None, axis=-1):
+            dtype = _dtype
+            if dtype is None:
+                dtype = fft_func(np.ones(8, dtype=a.dtype)).dtype
+
+            if len(a.chunks[axis]) != 1:
+                raise ValueError(chunk_error % (axis, a.chunks[axis]))
+
+            s = n
+            if n is not None:
+                s = [None] * a.ndim
+                s[axis] = n
 
-        if len(a.chunks[axis]) != 1:
-            raise ValueError(chunk_error % (axis, a.chunks[axis]))
+            axes = (axis,)
 
-        chunks = out_chunk_fn(a, n, axis)
+            chunks = out_chunk_fn(a, s, axes)
 
-        return a.map_blocks(fft_func, n=n, axis=axis, dtype=dtype,
-                            chunks=chunks)
+            return a.map_blocks(fft_func, n=n, axis=axis, dtype=dtype,
+                                chunks=chunks)
```

Comment:
Yeah, I was thinking about that as well. Ran out of time to pursue it yesterday. So tabled it. Might try again today. The main thing is the arguments have different names. So we have to handle this somehow (maybe a second wrapper). I don't know if there are any other consequences of having a second wrapper like that.

Other than that we will have to move some of the branches into the function. Not sure whether it will be easier or harder to reason about what it is doing. At a bare minimum, branches for the 1-D and N-D cases next to each other should make it clear what the difference is, but there will be a few more of them. Some cases like guessing `dtype` and raising chunk errors can just use the N-D case for 1-D one.